src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.id

```diff
@@ -17,29 +17,63 @@ type StoragePowerActor struct {
         peerId        libp2p.PeerID  // TODO: will be removed likely (see: https://github.com/filecoin-project/specs/pull/555#pullrequestreview-300991681)
     ) addr.Address
 
-    // TODO add NotifyConsensusFault(addr.Address, [Block]) TODO maybe rename to Arbitrate
-    UpdatePower(address addr.Address, newPower StoragePower)
+    // TODO add NotifyConsensusFault(addr.Address, [block.Block]) TODO maybe rename to Arbitrate
+    UpdatePower(address addr.Address, newPower block.StoragePower)
 
-    TotalStorage() StoragePower
+    TotalStorage() block.StoragePower
 
-    PowerLookup(minerAddr addr.Address) StoragePower
+    PowerLookup(minerAddr addr.Address) block.StoragePower
 
     // call by StorageMinerActor on successful seal
     // TODO: workerKey of type StorageMiner.WorkerPubKey
     IncrementPower(minerAddress addr.Address)
 
     SuspendMiner(address addr.Address)
 
-    // UpdateStoragePowerSubsystem(Tipset, StateTree)
+    expectedConsensus ExpectedConsensus
+    ReportConsensusFault(
+        slasherAddr  addr.Address
+        faultType    ConsensusFaultType
+        proof        [block.Block]
+    )
+    ReportUncommittedPowerFault(cheaterAddr addr.Address, numSectors UVarint)
+    PowerTable
 
-    // VerifyElectionProof(base.Address, base.ElectionProof)	bool
+    // VerifyElectionProof(addr.Address, ElectionProof)	bool
 
     // TODO: old methods, should be removed
     // UpdateStoragePowerSubsystem(Tipset, StateTree)
-    // VerifyElectionProof(addr.Address, base.ElectionProof)	bool
+    // VerifyElectionProof(addr.Address, ElectionProof)	bool
     // NotifyStorageFault(addr.Address)
-    // TryPublishBlock(StateTree)					Block
+    // TryPublishBlock(StateTree)					block.Block
+    CommitPledgeCollateral(deals [deal.StorageDeal])
+    DecommitPledgeCollateral(deals [deal.StorageDeal])
+
+    pledgeCollateralBalances {addr.Address: UVarint}
+}
+
+type PowerTable struct {
+    // all power here is always verified
+    RegisterMiner(
+        addr        addr.Address
+        pk          filcrypto.PubKey
+        sectorSize  sector.SectorSize
+    )
+    miners {addr.Address: StorageMiner}
+    GetMinerPower(addr addr.Address) block.StoragePower
+    GetTotalPower() block.StoragePower
+    GetMinerPublicKey(addr addr.Address) filcrypto.PubKey
+    IncrementPower(addr addr.Address, numSectors UVarint)
+    SuspendPower(addr addr.Address, numSectors UVarint)
+    UnsuspendPower(addr addr.Address, numSectors UVarint)
+    RemovePower(addr addr.Address, numSectors UVarint)
+    RemoveAllPower(addr addr.Address, numSectors UVarint)
+}
 
-    // RegisterStorageMiner(SealProof)				Address
-    // TryUpdateStorage(base.Address, newPower)			error
+type StorageMiner struct {
+    MinerAddress         addr.Address
+    MinerStoragePower    block.StoragePower
+    MinerSuspendedPower  block.StoragePower
```

Comment:
Two things. I think passing address is better design since we would want to use the same methods for both the cronactor and self-reporting. So I chose to leave it there. Happy to change though.

My thought on suspended power is that it is necessary in order to account for collateral vs power. For instance if I declare a fault and my power drops, I should not be able to withdraw collateral. For that reason we need an intermediary state for power between just "present" and "absent", hence "suspended".

It may be worth our making a state machine for faults @zixuanzh